src/store/header/decodeJwtSlice.ts

```diff
@@ -0,0 +1,78 @@
+import {createSlice} from "@reduxjs/toolkit";
+import jwt_decode from "jwt-decode";
+
+type DecodeJWT = {
```

Comment:
Pls use nouns for type names, e.g.  `DecodedJWT`, not `DecodeJWT`. 
`decodeJWT` is a good name for a function, not for a type.

Reply:
Since the JWT holds the session info, maybe the type can be named appropriately:

```
type UserSession = {
```

usage:

```
const session: UserSession = jwt_decode(token);
```

This and simpler names for store slices and reducer properties:

```
const userSessionSlice = createSlice({
  name: "UserSession",
  initialState,
  reducers: {
    userName: (state) => { // or just `name:`
      state.name = decode().name;
    },
    userPhone: ...

```

Right now naming seems a bit complex (e.g. `nameDecodeUser`), and repeated use of `decode`/`decoded` won't add readability outside of this module.